core/gui/src/app/dashboard/component/user/user-workflow/user-workflow.component.ts

```diff
@@ -342,14 +342,20 @@ export class UserWorkflowComponent implements AfterViewInit {
   /**
    * Verify Uploaded file name and upload the file
    */
-  public onClickUploadExistingWorkflowFromLocal = (file: NzUploadFile): boolean => {
+  public onClickUploadExistingWorkflowFromLocal = (file: NzUploadFile): Observable<boolean> => {
     const fileExtensionIndex = file.name.lastIndexOf(".");
-    if (file.name.substring(fileExtensionIndex) === ".zip") {
-      this.handleZipUploads(file as unknown as Blob);
-    } else {
-      this.handleFileUploads(file as unknown as Blob, file.name);
-    }
-    return false;
+
+    const uploadPromise = (async () => {
+      if (file.name.substring(fileExtensionIndex) === ".zip") {
+        this.handleZipUploads(file as unknown as Blob);
+      } else {
+        this.handleFileUploads(file as unknown as Blob, file.name);
+      }
+      await this.search(true);
+      return false;
+    })();
+
+    return from(uploadPromise);
```

Comment:
Please add a notification message saying that the workflow has been uploaded successfully